### ISSUE TITLE
Inner class annotated with @Parcel causes org.parceler.transfuse.util.TransfuseRuntimeException 

### ISSUE THREAD

**Issue by justincpollard** [johncarl81/parceler#82]
Hi John,

Related to https://github.com/johncarl81/parceler/issues/75.

We've been seeing a few crashes caused by `org.parceler.ParcelerRuntimeException: Unable to create ParcelableFactory for ...`. In an attempt to make sure that these don't pop up, I've added all `@Parcel` annotated classes to the `@ParcelClasses` array. One of our `@Parcel` classes is an inner class of an `@Fragment` class and throws the following at code gen time:

`Exception in thread "pool-743-thread-1" org.parceler.transfuse.util.TransfuseRuntimeException: Unable to perform code generation`

Which is caused by:

`Caused by: javax.annotation.processing.FilerException: Attempt to recreate a file for type com.example.MyClass$MyInnerClass$$Parcelable`.

It appears that removing the inner class from the `@ParcelClasses` fixes the issue, but I'm wondering if you have any guidance/best practices for which `@Parcel` classes to add to the `@ParcelClasses` array.

**Comment by johncarl81** [comment]
That is strange.  Is the inner class annotated with `@Parcel`?

**Comment by justincpollard** [comment]
Yea, the inner class is annotated with `@Parcel` as follows:

```java
@Parcel
  public static class IdentityDialogState {
    public Long agentId;
    public String agentName;
    public String agentEmail;

    @ParcelConstructor
    public IdentityDialogState(Long agentId, String agentName, String agentEmail) {
      this.agentId = agentId;
      this.agentName = agentName;
      this.agentEmail = agentEmail;
    }
  }
```

**Comment by johncarl81** [comment]
If you have the inner class (`IdentifyDialogState`) annotated with `@Parcel` and it is also in the `@ParcelClasses` list then it will try to create the `Parcelable` twice, which is my guess why you're seeing the code generation exception.  Could you remove `@Parcel`  from `IdenfityDialogState`?

**Comment by justincpollard** [comment]
Ok, great! I wasn't aware that I should use only `@Parcel` or `@ParcelClasses`, but not both. I think that solves my issue.

**Comment by johncarl81** [comment]
Really though, you should be able to either annotate the class with `@Parcel` or reference it with `@ParcelClass`.  Any idea why the `Parcelable` isn't being generated when it is annotated with `@Parcel`?  Is this class in a library?

**Comment by justincpollard** [comment]
My initial question probably wasn't clear in this regard. It wasn't `IdentityDialogState` that caused the `org.parceler.ParcelerRuntimeException: Unable to create ParcelableFactory for ...`; it was a different class and you're right to think this class is in a library; it is! In fact, nearly all of our `@Parcel` classes are library classes, which (I think) makes ensuring that they are generated a little difficult. That's why I've opted for the 'kitchen sink' approach adding them all to `@ParcelClasses`.

Something still puzzles me, though. Even after I've added all `@Parcel` library classes to `@ParcelClasses` (save `IdentityDialogState`, which was the only non-library `@Parcel` class) I don't see the error. 

Initially, we had assumed that if we have 2 classes - `ParcelClassA` and `ParcelClassB` - and that `ParcelClassA` has a member of type `ParcelClassB`, the appropriate `ParcelClassB` parceler code would be generated at the time that the `ParcelClassA` parceler code is generated. If what we've assumed is true, why don't we see the same `Caused by: javax.annotation.processing.FilerException: Attempt to recreate a file for type com.example.MyClass$MyInnerClass$$Parcelable` for `ParcelClassB` when it is, in fact, in the `@ParcelClasses` array as well as annotated with `@Parcel`?
